### PR TITLE
Feat/table on row and cell

### PR DESCRIPTION
**Pull request by qinhuaihe**
### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link
#1173 

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [ ] Changelog is provided or not needed

### CODE REVIEW

**Comment by qinhuaihe** [comment]
I have not finish the  `OnHeaderRow` and `OnHeaderCell` methods yet.

**Comment by qinhuaihe** [comment]
I'm sorry for pushing the irrelevant old commits. I tried to fix it, but failed.

**Comment by qinhuaihe** [comment]
Look at this [line](https://github.com/qinhuaihe/ant-design-blazor/blob/feat/table-on-row-and-cell/components/table/Column.razor.cs#L54). I can't use the ItemType as the generic parameter of RowData, so I have to cast the rowData when use it. Can I add another typeparam to Column?

```
Dictionary<string, object> OnCell(RowData rowData)
    {
        // have to cast rowData type
        var row = (RowData<Data>)rowData;

        Action<MouseEventArgs> OnClick = args =>
        {
            Console.WriteLine($"cell {row.Data.Name} was clicked");
        };

        return new Dictionary<string, object>()
        {
            { "onclick", OnClick },
        };
    }
```

File: site/AntDesign.Docs/Demos/Components/Table/demo/on-row-and-on-cell.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
+---
+order: 1
+title:
+  en-US: OnRow usage
+  zh-CN: OnRow �÷�
+---
+
+## zh-CN
+
+������ `OnRow` `OnHeaderRow` `OnCell` `OnHeaderCell`��
```
Comment (ElderJames) [inline site/AntDesign.Docs/Demos/Components/Table/demo/on-row-and-on-cell.md:10]: please fix the encoding

**Review by ElderJames — CHANGES_REQUESTED** [review]

**Comment by qinhuaihe** [comment]
Please review the pr, we need this functionality urgently. Thank you in advance. 😀

**Review by ElderJames — APPROVED** [review]

**Pull request merged**